### PR TITLE
Revert the revert of "Hide the window until we're finished with initialization"

### LINKED ISSUE #11561: Starting WT momentarily shows a black titlebar
### Windows Terminal version (or Windows build number)

10.0.19043.1288 1.12.2922.0

### Other Software

_No response_

### Steps to reproduce

Start WT.

### Expected Behavior

The GUI should start without showing a glitched black title bar, just as it did in v1.11.

### Actual Behavior

It shows a black title bar for a few milliseconds. See below:

My [settings.json](https://github.com/rashil2000/dotfiles/blob/master/wt/settings.json) is the same as before I upgraded (through Store).

### PR DESCRIPTION
**Pull request by zadjii-msft**
This is a revert of the revert of #12979. We mainly reverted that PR because of an issue where restored windows would grow/shrink slightly on external displays. It was too close to the ship date for that release, so we backed it out wholesale (in #13098). I think I've found the real root of the problem, and fixed it here. 

The money diff here from the original PR: 4c08b9a1bc2e90b8284e4d8117d0de400784520f...c34495dcfc19ea67a9f4f9673d422760200683ab. Basically, I had put the part where we actually handle the creation of the window into `_AppInitializedHandler`, when we should have left the window to be created in `_HandleCreateWindow` We create it there, _hidden_, and then should only _show_ it in `_AppInitializedHandler`. 

I'm _NOT_ incorporating the change for #9053. I reverted that bit in 1fac40355. I am too worried about that messing with the phwnd that I wanted to get that reviewed and committed atomically, separately. 

* fixes  #11561
* tested manually
* I work here

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -704,10 +704,55 @@ namespace winrt::TerminalApp::implementation
...
+            {
+                // We don't exist anymore? Well, we probably don't need to fire
+                // off an Initialized event then...
+                co_return;
+            }
+
+            if (auto self{ weak.get() })
+            {
+                self->_InitializedHandlers(*self, nullptr);
+            }
```
Comment (lhecker) [inline src/cascadia/TerminalApp/TerminalPage.cpp:755]: I think something like this is equivalent but much shorter:
```suggestion
            Dispatcher().RunAsync(CoreDispatcherPriority::Low, [weak = get_weak()]() {
                if (auto self{ weak.get() })
                {
                    self->_InitializedHandlers(*self, nullptr);
                }
            });
```

**Review by lhecker — APPROVED** [review]

**Comment by DHowett** [comment]
@msftbot make sure @winobjc-bot signs off

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 1 above)
Comment (DHowett) [reply src/cascadia/TerminalApp/TerminalPage.cpp:755]: Yoo I agree let's do it. Sometimes coroutines are the answer! Sometimes they are not.

**Review by DHowett — APPROVED** [review]
This seems inoffensive and straightforward. Is it going to explode? How can we be sure?

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 1 above)
Comment (DHowett) [reply src/cascadia/TerminalApp/TerminalPage.cpp:755]: (This also compiles to fewer coroutine funclets!)

**Comment by DHowett** [comment]
On second thought perhaps we should hold this for 1.17+?

**Comment by zadjii-msft** [comment]
Yea maybe. Going back to 4c08b9a didn't fix it either. Uhg.

**Comment by zadjii-msft** [comment]
Huh, I never added real notes as to why we held this one out. IIRC it was something like the fact that this would consistently work for the _first_ window I'd make but not subsequent ones, which is _w e i r d_


And now that I'm 6 months removed from that code, that makes me thing it might have something to do with summoning peasant windows after creation, that the monarch wouldn't do. That's at least something that would be obviously different between the N=0 and N>=1 windows.

**Comment by zadjii-msft** [comment]
@Nirgranth if you'd like to help out, I'd start here. Or possibly even further back in the past, to the original version of this PR, #12979. 

The original idea was that we shouldn't call `ShowWindow` in our `WM_CREATE` handler, we should instead wait until the XAML content is fuly initialized to show the window. 

That original PR had some issue about getting the initial sizing just a little wrong, so we reverted that. This PR attempted to fix that sizing issue (and definitely did figure the sizing thing out).

But somewhere between there (#12979) and now, this PR doesn't seem to work at all anymore. Probably an incoming merge from `main`, but there's months of commits here now. It might just be best to start from the original idea fresh again. 

Anyways, that's my notes. I'd doubt I'll have time to loop back on this one soon, so any help would be appreciated. I'm guessing there's a rogue `ShowWindow` call that's happening, but I don't see anything obvious to blame.

**Comment by zadjii-msft** [comment]
Magically, this works again after 6a99f25. We needed another jiggle the handle to let XAML paint

File: src/cascadia/WindowsTerminal/WindowEmperor.cpp [hunk 2]
```diff
@@ -169,18 +169,13 @@ void WindowEmperor::_windowStartedHandlerPostXAML(const std::shared_ptr<WindowTh
...
-    args.WindowID(sender->Peasant().GetID());
-    args.SummonBehavior().MoveToCurrentDesktop(false);
-    args.SummonBehavior().ToggleVisibility(false);
-    args.SummonBehavior().DropdownDuration(0);
-    args.SummonBehavior().ToMonitor(Remoting::MonitorBehavior::InPlace);
-    _manager.SummonWindow(args);
+    // If you summon here, the resulting code will call ShowWindow(SW_SHOW) on
+    // the Terminal window, making it visible BEFORE the XAML island is actually
+    // ready to be drawn. We want to wait till the app's Initialized event
+    // before we make the window visible.
```
Comment (DHowett) [inline src/cascadia/WindowsTerminal/WindowEmperor.cpp:178]: I'm a bit worried that (1) we used to Summon from here and (2) now we aren't. Will this result in new terminal windows not coming to the front when they are the second window in the process? Will this result in us missing some activations somehow?

**Review by DHowett — COMMENTED** [review]
Could this PR be the cause of #15139...somehow?

**Pull request merged**